src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
> So if we're going to do that we should just set CURLOPT_SSL_VERIFYHOST to 0 and always do it ourselves.

Remember, this is the slow path... by the time we get here, either there was a server callback registered so we didn't try to use OpenSSL's verification, or OpenSSL's verification failed.  The common case is we use OpenSSL's verification and it succeeds.  We do not want to force to the slow path every time, as it's, well, slow, compared to OpenSSL's verification.